routes/users.js

```diff
@@ -1,272 +1,277 @@
-        
-        // update user and add aboutMe
-        User.updateOne(
-            {_id: candidate._id}, 
-            {aboutMe : aboutMe },
-            {multi:true}, 
-              function(err, numberAffected){  
-              });
+    const candidate = await User.findOne({ _id: req.session.userId });
 
-        res.status(200).send(JSON.stringify({
-            success: true,
-            error: null,
-            data: aboutMe,
-        }));
+    // candidate with id not found in database, return not found status 404
+    if (!candidate) {
+      res.status(404).send(
+        JSON.stringify({
+          success: false,
+          error: "User could not be found",
+        })
+      );
+    }
 
-	} catch (err) {
-		res.status(400).send(JSON.stringify({
-			success: false,
-			error: err
-		}));
-	}
-});
+    // update user and add aboutMe
+    User.updateOne(
+      { _id: candidate._id },
+      { aboutMe: aboutMe },
+      { multi: true },
+      function (err, numberAffected) {}
+    );
 
+    res.status(200).send(
+      JSON.stringify({
+        success: true,
+        error: null,
+        data: aboutMe,
+      })
+    );
+  } catch (err) {
+    res.status(400).send(
+      JSON.stringify({
+        success: false,
+        error: err,
+      })
+    );
+  }
+});
 
 // @desc    Render agency.ejs
 // @route   GET '/users/agency'
 // @access  Private, only userRole == partners
 // @tested 	No
-router.get('/agency', redirectLogin, async (req, res) => {
-	try {
-		res.render('agency', {
-			user: res.locals.user
-		});
-	} catch (err) {
-		res.status(400).send(JSON.stringify({
-			success: false,
-			error: err
-		}));
-	}
+router.get("/agency", redirectLogin, async (req, res) => {
+  try {
+    res.render("agency", {
+      user: res.locals.user,
+    });
+  } catch (err) {
+    res.status(400).send(
+      JSON.stringify({
+        success: false,
+        error: err,
+      })
+    );
+  }
 });
 
-
 // @desc    agency info is sent to db
 // @route   POST '/users/agency'
 // @access  private, partners only
 // @tested 	No
-router.post('/agency', async (req, res) => {
-	const {
-		agencyName,
-		agencyWebsite,
-		agencyPhone,
-		agencyBio
-	} = req.body;
+router.post("/agency", async (req, res) => {
+  const { agencyName, agencyWebsite, agencyPhone, agencyBio } = req.body;
 
-	const newAgency = new Agency({
-		agencyName,
-		agencyWebsite,
-		agencyPhone,
-		agencyBio
-	});
-	try {
-		await newAgency.save();
-		console.log("agency data saved");
-		return res.send('/users/profile');
-	} catch (err) {
-		console.log(err);
-	}
+  const newAgency = new Agency({
+    agencyName,
+    agencyWebsite,
+    agencyPhone,
+    agencyBio,
+    accountManager: req.session.userId,
+  });
+  try {
+    await newAgency.save();
+    var agencyId = mongoose.Types.ObjectId(newAgency._id);
+    req.session.agencyId = agencyId;
+    console.log("agency data saved");
+    return res.send("/users/profile");
+  } catch (err) {
+    console.log(err);
```

Comment:
Without a response I think as soon as this fails the client gets a timeout? we should at least throw a 500 here